sequencer.c

```diff
@@ -5603,10 +5605,101 @@ static int skip_unnecessary_picks(struct repository *r,
 	return 0;
 }
 
+struct todo_add_branch_context {
+	struct todo_list new_list;
+	struct strbuf *buf;
+};
+
+static int add_branch_for_decoration(const struct name_decoration *d, void *data)
+{
+	struct todo_add_branch_context *ctx = data;
+	size_t base_offset = ctx->buf->len;
+	int i = ctx->new_list.nr;
+	struct todo_item *item;
+	const char *branch_name;
+
+	/* Do not try updating a branch that is checked out. */
+	if (branch_checked_out(d->name, NULL))
```

Comment:
This should probably warn, saying _where_ it is checked out. And offering a command-line how it could be updated (`git -C <worktree> reset --keep <commit>` or something like that).

Reply:
I thought about that, except that the tip ref is (usually) checked out in the current worktree, which makes this more complicated.

Reply:
I'm just thinking about the user experience here, and I think I, as a user, would appreciate that extra mile.

Reply:
Perhaps this could be added as a follow-up for polish.

Reply:
I figured out how to add a comment in the TODO list, so that's what I'm going with.